sky/jobs/utils.py

```diff
@@ -257,7 +257,22 @@ def _handle_legacy_job(job_id: int):
 
         # For jobs with schedule state:
         pid = tasks[0]['controller_pid']
-        if pid is None:
+        if schedule_state == managed_job_state.ManagedJobScheduleState.DONE:
+            if all(task['status'].is_terminal() for task in tasks):
+                # Turns out this job is fine, even though it got pulled by
+                # get_jobs_to_check_status. Either the job was somehow in an
+                # inconsistent state that already resolved itself, or we have
+                # a bug in the query.
```

Comment:
would it be possible to be a concurrency issue for fetching vs checking here, i.e. the job is normally updated to terminal state, after the `get_jobs_to_check_status` is called?

Reply:
It shouldn't normally happen, because terminals status is always updated _before_ setting the schedule_stare to DONE. The only way I can think is if another invocation of this function is running in parallel.